cli/flox-rust-sdk/src/models/environment/remote_environment.rs

```diff
@@ -98,20 +102,62 @@ impl RemoteEnvironment {
         )
         .unwrap();
 
-        let inner_out_link = gcroots_dir(flox, &pointer.owner).join(remote_branch_name(&pointer));
-        let mut inner =
-            ManagedEnvironment::open_with(floxmeta, flox, pointer, dot_flox_path, inner_out_link)
-                .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
+        let inner_rendered_env_paths = {
+            let gcroots_dir = gcroots_dir(flox, &pointer.owner);
+            if !gcroots_dir.exists() {
+                std::fs::create_dir_all(&gcroots_dir)
+                    .map_err(RemoteEnvironmentError::CreateGcRootDir)?;
+            }
+            let base_dir =
+                CanonicalPath::new(gcroots_dir).expect("gcroots_dir is not a valid path");
+
+            RenderedEnvironmentLinks::new_in_base_dir_with_name_and_system(
+                &base_dir,
+                remote_branch_name(&pointer),
+                &flox.system,
+            )
+        };
+
+        let mut inner = ManagedEnvironment::open_with(
+            floxmeta,
+            flox,
+            pointer.clone(),
+            dot_flox_path,
+            inner_rendered_env_paths,
+        )
+        .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
 
         // (force) Pull latest changes of the environment from upstream.
         // remote environments stay in sync with upstream without providing a local staging state.
         inner
             .pull(flox, true)
             .map_err(RemoteEnvironmentError::ResetManagedEnvironment)?;
 
-        let out_link = path.join(GCROOTS_DIR_NAME);
+        let rendered_env_paths = {
+            let gcroots_dir = path.join(GCROOTS_DIR_NAME);
+
+            // TODO: current .flox/run is _a link_!
+            // Migration to a new environment by deleting the link and creating dir for _multiple_ links
```

Comment:
We should make sure to test migration, manually or with a test

Reply:
added a unit test